### DISCUSSION TITLE
Question about rules set by config profiles

### DISCUSSION THREAD

**Discussion by cricketer94** [usnistgov/macos_security#440] [Q&A] [answered]
Hi, 

Have been using these rules for a while, and have noticed quite a few rules where the check is set by a config profile (and the script seems to be using Javascript to check whether these config profiles exist). An example is this the system_settings_firewall_enable rule. 

Are there ways for us to check whether the control is enabled if we are not using an MDM profile to enforce this control (maybe due to cross-platform compatability on Intune e.g.). Or do we ignore those Javascript controls for these checks? 

Thank you

**Comment by robertgendler** [comment, marked as answer]
@cricketer94 

So the reason everything is set the way they are is due to the fact most rules need to be enforced. Certain settings can't be enforced when not set with a configuration profile, they would just sort of be enabled. Some settings can be checked with the same JXA check we have in place for profiles but some can not. For those that you'd rather they aren't set by profile and the JXA check does not work, you would have to create your own custom check for the rule. 

As far as I know, things should work the same when set through inTune vs Jamf vs Mosyle vs Addigy vs Joes Chicken Shack MDM vs whatever and done with a configuration profile.

Hope that answers the question.
